Preparing change log for beta release (#147)
silverstripe/developer-docs: en/06_Project_Governance/06_Supported_Modules.md

---
title: Supported modules
---

# Commercially supported modules for Silverstripe CMS 5

> Silverstripe CMS ships modules that receive commercial support. Commercially Supported modules receive regular updates to work with the latest Silverstripe CMS release. Their APIs conform to semantic version. They are covered by:

Author: ```suggestion
Silverstripe CMS ships modules that receive commercial support. Commercially supported modules receive regular updates to work with the latest Silverstripe CMS release. Their APIs conform to semantic version. They are covered by:
```

[silverstripe/versionfeed](https://packagist.org/packages/silverstripe/versionfeed) | 3
[silverstripe/webauthn-authenticator](https://packagist.org/packages/silverstripe/webauthn-authenticator) | 5
[symbiote/silverstripe-advancedworkflow](https://packagist.org/packages/symbiote/silverstripe-advancedworkflow) | 6
[symbiote/silverstripe-gridfieldextensions](https://packagist.org/packages/symbiote/silverstripe-gridfieldextensions) | 4
[symbiote/silverstripe-multivaluefield](https://packagist.org/packages/symbiote/silverstripe-multivaluefield) | 6
[symbiote/silverstripe-queuedjobs](https://packagist.org/packages/symbiote/silverstripe-queuedjobs) | 5
[tractorcow/silverstripe-fluent](https://packagist.org/packages/tractorcow/silverstripe-fluent) | 7

## Supported NPM packages

The following two NPM packages are also supported because they are required to build the UI of Silverstripe CMS 5:
- [@silverstripe/webpack-config](https://www.npmjs.com/package/@silverstripe/webpack-config)
- [@silverstripe/eslint-config](https://www.npmjs.com/package/@silverstripe/eslint-config)

## Other modules in the "silverstripe" namespace

> They are other modules hosted under the _silverstripe_ Packagist namespace. These modules are maintained on a best effort basis. They do not go through regular regression testing. Their APIs may be more fluid than supported modules. They maybe more experimental or may not receive the same level of care as supported modules.

Reviewer: ```suggestion
There are other modules hosted under the _silverstripe_ Packagist namespace. These modules are maintained on a best effort basis. They do not go through regular regression testing. Their APIs may be more fluid than supported modules. They maybe more experimental or may not receive the same level of care as supported modules.
```
Sorry, should have caught this the first time around.

Author: Fixed.

> They are other modules hosted under the _silverstripe_ Packagist namespace. These modules are maintained on a best effort basis. They do not go through regular regression testing. Their APIs may be more fluid than supported modules. They maybe more experimental or may not receive the same level of care as supported modules.

Reviewer: ```suggestion
They are other modules hosted under the _silverstripe_ Packagist namespace. These modules are maintained on a best effort basis and aren't guaranteed to go through regular regression testing. Their APIs may be more fluid than supported modules. They maybe more experimental or may not receive the same level of care as supported modules.
```

Author: fixed.

---

> title: Supported modules

Reviewer: ```suggestion
title: Supported modules
description: Modules which are commercially supported by Silverstripe
```

Author: Done

Author: Done.

---

> # Commercially supported modules for Silverstripe CMS 5

Reviewer: ```suggestion
# Commercially supported modules
```
No need to say which version it's for - the docs site already does that. This is just another heading we'd have to update next major for no benefit.

> Silverstripe CMS ships modules that receive commercial support. Commercially supported modules receive regular updates to work with the latest Silverstripe CMS release. Their APIs conform to semantic version. They are covered by:

Reviewer: ```suggestion
Silverstripe CMS ships modules that receive commercial support. Commercially supported modules receive regular updates to work with the latest Silverstripe CMS release. Their APIs conform to [semantic versioning](https://semver.org/). They are covered by:
```

> - our [security release process](./Managing_Security_Issues) and 

Reviewer: ```suggestion
- our [security release process](Managing_Security_Issues) and 
```
It's already relative to the current location without using `./`, and we don't tend to use `./` elsewhere in the docs.
This applies to any other links you've done this way in this PR.

- our [major release policy](./Major_release_policy).

These modules will be supported for the lifetime of Silverstripe CMS 5 in the provided major versions.

## Core Silverstripe CMS modules

These modules provide the core Silverstripe CMS experience. Most Silverstripe CMS projects should install all of them.

> Core Module | Supported versions

Reviewer: ```suggestion
Core Module | Supported major release line
```

Reviewer: `2` is not a version, it's a line of versions.
This change also applies to the other table below.

If you have a good reason to not make this change, then please use `version` instead of `versions`.

-- | --

> [silverstripe/admin](https://packagist.org/packages/silverstripe/admin) | 2

Reviewer: There is a mix of github and packagist links here. Better to just stick to one or the other.

Author: Should all be using Packagist now.

[silverstripe/asset-admin](https://packagist.org/packages/silverstripe/asset-admin) | 2
[silverstripe/assets](https://packagist.org/packages/silverstripe/assets) | 2
[silverstripe/cms](https://packagist.org/packages/silverstripe/cms) | 5
[silverstripe/config](https://github.com/silverstripe/silverstripe-config) | 2
[silverstripe/campaign-admin](https://packagist.org/packages/silverstripe/campaign-admin) | 2
[silverstripe/errorpage](https://packagist.org/packages/silverstripe/errorpage) | 2
[silverstripe/framework](https://packagist.org/packages/silverstripe/framework) | 5
[silverstripe/graphql](https://packagist.org/packages/silverstripe/graphql) | 5
[silverstripe/installer](https://github.com/silverstripe/silverstripe-installer) | 5
[silverstripe/login-forms](https://packagist.org/packages/silverstripe/login-forms) | 5
[silverstripe/mimevalidator](https://packagist.org/packages/silverstripe/mimevalidator) | 3
[silverstripe/recipe-cms](https://github.com/silverstripe/recipe-cms) | 5
[silverstripe/recipe-core](https://github.com/silverstripe/recipe-core) | 5

> [silverstripe/recipe-plugin](https://github.com/silverstripe/recipe-plugin) | 2

Reviewer: recipe-plugin and vendor-plugin aren't modules.... though I don't know if we need to make this distinction. We probably should since we made a distinction for the npm packages.
It might even make sense to distinguish recipes - though recipes are much more similar to modules than composer plugins are so I'm okay with keeping those lumped in if that's what you prefer.

Author: Honestly, I think it's a distinction without a difference ... and it saves me drafting a dissertation about the difference between a module, a package and a plugin.

Reviewer: For the composer plugins it's almost as much of a difference as the npm packages which you've separated out - but I'm fine with leaving these as is, it's not a big deal. Just my brain being pedantic.

[silverstripe/reports](https://packagist.org/packages/silverstripe/reports) | 5
[silverstripe/session-manager](https://packagist.org/packages/silverstripe/session-manager) | 2
[silverstripe/siteconfig](https://packagist.org/packages/silverstripe/siteconfig) | 5
[silverstripe/vendor-plugin](https://github.com/silverstripe/vendor-plugin) | 2
[silverstripe/versioned](https://packagist.org/packages/silverstripe/versioned) | 2
[silverstripe/versioned-admin](https://packagist.org/packages/silverstripe/versioned-admin) | 2

> ## Optional modules

Reviewer: All modules are optional.... I'd prefer "Other supported modules" here. But you can take or leave this suggestion as it's highly subjective.

[silverstripe/userforms](https://packagist.org/packages/silverstripe/userforms) | 6
[silverstripe/versionfeed](https://packagist.org/packages/silverstripe/versionfeed) | 3
[silverstripe/webauthn-authenticator](https://packagist.org/packages/silverstripe/webauthn-authenticator) | 5
[symbiote/silverstripe-advancedworkflow](https://packagist.org/packages/symbiote/silverstripe-advancedworkflow) | 6
[symbiote/silverstripe-gridfieldextensions](https://packagist.org/packages/symbiote/silverstripe-gridfieldextensions) | 4
[symbiote/silverstripe-multivaluefield](https://packagist.org/packages/symbiote/silverstripe-multivaluefield) | 6
[symbiote/silverstripe-queuedjobs](https://packagist.org/packages/symbiote/silverstripe-queuedjobs) | 5
[tractorcow/silverstripe-fluent](https://packagist.org/packages/tractorcow/silverstripe-fluent) | 7

## Supported NPM packages

The following two NPM packages are also supported because they are required to build the UI of Silverstripe CMS 5:
- [@silverstripe/webpack-config](https://www.npmjs.com/package/@silverstripe/webpack-config)
- [@silverstripe/eslint-config](https://www.npmjs.com/package/@silverstripe/eslint-config)

> You'll only need to interact with them if you are customising the Silverstripe CMS UI.

Reviewer: This sentence seems to suggest that customising the UI is uncommon, which it isn't.
I'd honestly recommend just leaving this sentence out altogether, but if you want to keep it, it should probably sound less opinionated and link to the relevant docs.

Author: Be gone sentence!

> ## Other non-supported modules

Reviewer: ```suggestion
## Other modules in the silverstripe namespace
```

> They are other modules hosted under the _silverstripe_ Packagist handle. These modules are maintained on a best effort basis. They do not go through regular regression testing. Their APIs may be more fluid than supported modules. They maybe more experimental or may not receive the same level of care as supported modules.

Reviewer: ```suggestion
They are other modules hosted under the _silverstripe_ Packagist namespace. These modules are maintained on a best effort basis. They do not go through regular regression testing. Their APIs may be more fluid than supported modules. They maybe more experimental or may not receive the same level of care as supported modules.
```
or organisation, perhaps - but I don't think handle is correct.

> They are other modules hosted under the _silverstripe_ Packagist handle. These modules are maintained on a best effort basis. They do not go through regular regression testing. Their APIs may be more fluid than supported modules. They maybe more experimental or may not receive the same level of care as supported modules.

Reviewer: This completely ignores the fact that some are maintained by members of bespoke teams, and _could_ be maintained by community members. I think we should add something about that in here - especially since the changelog explicitly asks people to let us know if they want to maintain some of these.

Maybe we should have a way to identify modules which have maintainers (list them in the readme perhaps?)

Author: I don't think we need to discuss the internal structure of Silverstripe in the official docs. I don't think there's anything in the previous module that wouldn't apply to a bespoke module. e.g.: TerraFormers could decide that they don't give a toss about the embargo module tomorrow, or they could tag a new major release, or they could could decide to fix a security issue without issuing a CVE.

Reviewer: It's not about the internal structure of Silverstripe, we're inviting community members to maintain some of these modules, and frankly anyone not in the cms squad is effectively "the community" from the standpoint of our documentation. So saying anything in the silverstripe github org which isn't a supported module "do not go through regular regression testing" basically is saying that community maintainers of those repos aren't doing much... which we can't really say. I'll recommend a specific change in a separate comment.

Author: I've added the "no guarantee of regression testing" bit as suggested.

> Unsupported modules can still be used in Silverstripe CMS projects, but should be considered as community modules.

Reviewer: ```suggestion
These modules can still be used in Silverstripe CMS projects, but should be considered as community modules.
```

---
title: Supported modules
---

# Supported modules

> These modules will be supported for the lifetime of Silverstripe CMS 5 in the provided versions.

Reviewer 2: ```suggestion
These modules will be supported for the lifetime of Silverstripe CMS 5 in the provided major versions.
```
